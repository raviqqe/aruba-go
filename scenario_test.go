package aruba_test

import (
	"bytes"
	"os"
	"runtime"
	"testing"

	"github.com/cucumber/godog"
	"github.com/cucumber/godog/colors"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/raviqqe/aruba-go"
	"github.com/stretchr/testify/assert"
)

func TestSuccessfulFeatures(t *testing.T) {
	status := godog.TestSuite{
		Name:                "aruba",
		ScenarioInitializer: aruba.InitializeScenario,
		Options: &godog.Options{
			Concurrency: runtime.NumCPU(),
			Output:      colors.Colored(os.Stdout),
			Format:      "pretty",
		},
	}.Run()

	assert.Zero(t, status)
}

func TestFailedFeatures(t *testing.T) {
	b := bytes.NewBuffer(nil)
	status := godog.TestSuite{
		Name:                "aruba",
		ScenarioInitializer: aruba.InitializeScenario,
		Options: &godog.Options{
			Concurrency: runtime.NumCPU(),
			Format:      "pretty",
			Output:      b,
			Paths:       []string{"failures"},
		},
	}.Run()

	assert.NotZero(t, status)
	snaps.MatchSnapshot(t, b.String())
}
