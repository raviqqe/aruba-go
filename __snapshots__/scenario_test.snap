
[TestFailedFeatures - 1]
Feature: Command

  Scenario: Run a command successfully # failures/command.feature:3
    When I successfully run `false`
    exit status 1 (stderr: "")

  Scenario: Check an exit status         # failures/command.feature:6
    When I run `true`
    Then the exit status should not be 0
    expected exit code 0 not to be 0 (stderr: "")

  Scenario: Check an exit status of 1    # failures/command.feature:10
    When I run `false`
    Then the exit status should not be 1
    expected exit code 1 not to be 1 (stderr: "")

  Scenario: Check an exit status of 0 # failures/command.feature:14
    When I run `true`
    Then the exit status should be 1
    expected exit code 0 to be 1 (stderr: "")

  Scenario: Run a command interactively  # failures/command.feature:18
    When I run `echo` interactively
    Then the exit status should not be 0
    expected exit code 0 not to be 0 (stderr: "")

  Scenario: Output stderr on failure # failures/command.feature:22
    When I run the following script:
      """ sh
      echo foo >&2
      exit 1
      """
    Then the exit status should be 0
    expected exit code 1 to be 0 (stderr: "foo\n")

Feature: File

  Scenario: Create a file with its content               # failures/file.feature:3
    Given a file named "foo.txt" with "foo"
    When the file named "foo.txt" should exist
    Then a file named "foo.txt" should not contain "foo"
    expected file "foo.txt" not to contain "foo"

  Scenario: Create a file without its content        # failures/file.feature:8
    Given a file named "foo.txt" with ""
    When the file named "foo.txt" should exist
    Then a file named "foo.txt" should contain "foo"
    expected file "foo.txt" to contain "foo"

  Scenario: Create a file with a doc-string              # failures/file.feature:13
    Given a file named "foo.txt" with:
      """
      foo
      """
    When the file named "foo.txt" should exist
    Then a file named "foo.txt" should not contain "foo"
    expected file "foo.txt" not to contain "foo"

  Scenario: Create a file with an empty doc-string   # failures/file.feature:21
    Given a file named "foo.txt" with:
      """
      
      """
    When the file named "foo.txt" should exist
    Then a file named "foo.txt" should contain "foo"
    expected file "foo.txt" to contain "foo"

  Scenario Outline: Check file existence                 # failures/file.feature:28
    Given a file named "foo.txt" with ""
    Then <article> file named "foo.txt" should not exist

    Examples:
      | article |
      | a       |
        file "foo.txt" should not exist
      | the     |
        file "foo.txt" should not exist

  Scenario: Create a directory                      # failures/file.feature:37
    Given a directory named "foo"
    Then the directory named "foo" should not exist
    directory "foo" should not exist

Feature: Standard I/O

  Scenario: Check stdout                     # failures/stdio.feature:3
    When I successfully run `echo foo`
    Then the stdout should not contain "foo"
    expected stdout "foo\n" not to contain "foo"

  Scenario: Check stderr                     # failures/stdio.feature:7
    When I run the following script:
      """ sh
      echo foo >&2
      """
    Then the stderr should not contain "foo"
    expected stderr "foo\n" not to contain "foo"

  Scenario: Check stdout with a blank character # failures/stdio.feature:14
    When I successfully run `echo foo`
    Then the stdout should not contain "\n"
    expected stdout "foo\n" not to contain "\n"

  Scenario: Check stdout to contain nothing # failures/stdio.feature:18
    When I successfully run `echo foo`
    Then the stdout should not contain ""
    expected stdout "foo\n" not to contain ""

  Scenario: Check stdout to contain exactly nothing # failures/stdio.feature:22
    When I successfully run `echo`
    Then the stdout should not contain exactly ""
    expected stdout "\n" not to contain exactly ""

  Scenario: Check stdout to contain a string # failures/stdio.feature:27
    When I successfully run `echo foo`
    Then the stdout should not contain "foo"
    expected stdout "foo\n" not to contain "foo"

  Scenario: Check stdout to contain an exact string # failures/stdio.feature:31
    When I successfully run `echo foo bar`
    Then the stdout should contain exactly "foo"
    expected stdout "foo bar\n" to contain exactly "foo"

  Scenario: Check stdout not to contain a string # failures/stdio.feature:35
    When I successfully run `echo foo`
    Then the stdout should contain "bar"
    expected stdout "foo\n" to contain "bar"

  Scenario: Check stdout to contain a string # failures/stdio.feature:40
    When I successfully run `echo foo`
    Then the stdout should not contain:
      """
      foo
      """
    expected stdout "foo\n" not to contain "foo"

  Scenario: Check stdout to contain an exact string # failures/stdio.feature:47
    When I successfully run `echo foo bar`
    Then the stdout should contain exactly:
      """
      foo
      """
    expected stdout "foo bar\n" to contain exactly "foo"

  Scenario: Check stdout not to contain a string # failures/stdio.feature:54
    When I successfully run `echo foo`
    Then the stdout should contain:
      """
      bar
      """
    expected stdout "foo\n" to contain "bar"

--- Failed steps:

  Scenario: Run a command successfully # failures/command.feature:3
    When I successfully run `false` # failures/command.feature:4
      Error: exit status 1 (stderr: "")

  Scenario: Check an exit status # failures/command.feature:6
    Then the exit status should not be 0 # failures/command.feature:8
      Error: expected exit code 0 not to be 0 (stderr: "")

  Scenario: Check an exit status of 1 # failures/command.feature:10
    Then the exit status should not be 1 # failures/command.feature:12
      Error: expected exit code 1 not to be 1 (stderr: "")

  Scenario: Check an exit status of 0 # failures/command.feature:14
    Then the exit status should be 1 # failures/command.feature:16
      Error: expected exit code 0 to be 1 (stderr: "")

  Scenario: Run a command interactively # failures/command.feature:18
    Then the exit status should not be 0 # failures/command.feature:20
      Error: expected exit code 0 not to be 0 (stderr: "")

  Scenario: Output stderr on failure # failures/command.feature:22
    Then the exit status should be 0 # failures/command.feature:28
      Error: expected exit code 1 to be 0 (stderr: "foo\n")

  Scenario: Create a file with its content # failures/file.feature:3
    Then a file named "foo.txt" should not contain "foo" # failures/file.feature:6
      Error: expected file "foo.txt" not to contain "foo"

  Scenario: Create a file without its content # failures/file.feature:8
    Then a file named "foo.txt" should contain "foo" # failures/file.feature:11
      Error: expected file "foo.txt" to contain "foo"

  Scenario: Create a file with a doc-string # failures/file.feature:13
    Then a file named "foo.txt" should not contain "foo" # failures/file.feature:19
      Error: expected file "foo.txt" not to contain "foo"

  Scenario: Create a file with an empty doc-string # failures/file.feature:21
    Then a file named "foo.txt" should contain "foo" # failures/file.feature:26
      Error: expected file "foo.txt" to contain "foo"

  Scenario Outline: Check file existence # failures/file.feature:28
    Then a file named "foo.txt" should not exist # failures/file.feature:30
      Error: file "foo.txt" should not exist

  Scenario Outline: Check file existence # failures/file.feature:28
    Then the file named "foo.txt" should not exist # failures/file.feature:30
      Error: file "foo.txt" should not exist

  Scenario: Create a directory # failures/file.feature:37
    Then the directory named "foo" should not exist # failures/file.feature:39
      Error: directory "foo" should not exist

  Scenario: Check stdout # failures/stdio.feature:3
    Then the stdout should not contain "foo" # failures/stdio.feature:5
      Error: expected stdout "foo\n" not to contain "foo"

  Scenario: Check stderr # failures/stdio.feature:7
    Then the stderr should not contain "foo" # failures/stdio.feature:12
      Error: expected stderr "foo\n" not to contain "foo"

  Scenario: Check stdout with a blank character # failures/stdio.feature:14
    Then the stdout should not contain "\n" # failures/stdio.feature:16
      Error: expected stdout "foo\n" not to contain "\n"

  Scenario: Check stdout to contain nothing # failures/stdio.feature:18
    Then the stdout should not contain "" # failures/stdio.feature:20
      Error: expected stdout "foo\n" not to contain ""

  Scenario: Check stdout to contain exactly nothing # failures/stdio.feature:22
    Then the stdout should not contain exactly "" # failures/stdio.feature:24
      Error: expected stdout "\n" not to contain exactly ""

  Scenario: Check stdout to contain a string # failures/stdio.feature:27
    Then the stdout should not contain "foo" # failures/stdio.feature:29
      Error: expected stdout "foo\n" not to contain "foo"

  Scenario: Check stdout to contain an exact string # failures/stdio.feature:31
    Then the stdout should contain exactly "foo" # failures/stdio.feature:33
      Error: expected stdout "foo bar\n" to contain exactly "foo"

  Scenario: Check stdout not to contain a string # failures/stdio.feature:35
    Then the stdout should contain "bar" # failures/stdio.feature:37
      Error: expected stdout "foo\n" to contain "bar"

  Scenario: Check stdout to contain a string # failures/stdio.feature:40
    Then the stdout should not contain: # failures/stdio.feature:42
      Error: expected stdout "foo\n" not to contain "foo"

  Scenario: Check stdout to contain an exact string # failures/stdio.feature:47
    Then the stdout should contain exactly: # failures/stdio.feature:49
      Error: expected stdout "foo bar\n" to contain exactly "foo"

  Scenario: Check stdout not to contain a string # failures/stdio.feature:54
    Then the stdout should contain: # failures/stdio.feature:56
      Error: expected stdout "foo\n" to contain "bar"


24 scenarios (24 failed)
51 steps (27 passed, 24 failed)

---
